File: console/src/types/today.ts
export interface Correlation {
  used: number
  cap: number
}

export interface TodayRecommendation {
  id: string
  match: string
  market: string
  model_prob: number
  market_prob: number
  edge: number
  size: number
  confidence: string
  strategy: string
  close: string
  veto: string
  correlation: Correlation
  seed: number
}

export interface TodaySummary {
  date: string
  surfaced: number
  above_filter: number
  would_stake_usd: number
  exposure_pct: number
  exposure_cap_pct: number
}

export interface TodayResponse {
  summary: TodaySummary
  recommendations: TodayRecommendation[]
}

File: api/models/today.py
from __future__ import annotations

from pydantic import BaseModel


class Correlation(BaseModel):
    used: float
    cap: float


class TodayRecommendation(BaseModel):
    id: str
    match: str
    market: str
    model_prob: float
    market_prob: float
    edge: float
    size: float
    confidence: str
    strategy: str
    close: str
    veto: str
    correlation: Correlation
    seed: int


class TodaySummary(BaseModel):
    date: str
    surfaced: int
    above_filter: int
    would_stake_usd: float
    exposure_pct: float
    exposure_cap_pct: float


class TodayResponse(BaseModel):
    summary: TodaySummary
    recommendations: list[TodayRecommendation]
